roles/openshift_preflight/masters/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+---
+# determine if yum install of master pkgs will work
+- when:
+    - not containerized
+  block:
+
+    - name: main packages for enterprise
+      when:
+        - deployment_type == "openshift-enterprise"
+      check_yum_update:
+        packages:
+          - atomic-openshift
+          - atomic-openshift-clients
+          - atomic-openshift-master
```

Comment:
I would suggest using a variable to track the 'package prefix' here, with the prefix being 'origin' for 'origin' deployments and 'atomic-openshift' for ocp deployments. Then there can be a single task for the main packages instead of having a task for each deployment_type (or even just a single task in total).